docs/testing/9_performance.md

```diff
@@ -0,0 +1,76 @@
+# Performance
+
+If the battery of tests in a project starts taking too long to complete, here's a few tips to improve performance:
+
+# setUpTestData and setUpClass
+
+If you have several test_ methods in your TestCase class and all of them (or most of them) are using data created in the method "setUp", you should pass that data creation to the method setUpTestData. Here's some key points about setUpTestData:
+
+- It's not a method of unittest, but it's a django creation that was created to help speed up test execution.
+- It's invoked by setUpClass. So, if you also override setUpClass to do something else, setUpTestData will only be called when you invoke the super's setUpClass.
+- Will only execute once for all tests inside a TestCase class, which means you can place data creation operations there to optimize your tests performance. After running all tests, there will be a rollback for all data created in setUpTestData.
+- It's a classmethod, so don't just rename setUp to setUpTestData.
+
+Here's an example, consider this TestCase:
+
+```python
+class PostTestCase(TestCase):
+    def setUp(self):
+        self.url = reverse("post-url")
+        self.data = {"field": "value"}
+        self.model = factories.ModelFactory()
+
+    def test_success(self):
+        result = self.client.post(f"{self.url}", data=self.data)
+        self.assertEqual(result.status_code, 201)
+        self.assertEqual(models.Model.objects.all().count(), 2)
+
+    def test_failure(self):
+        result = self.client.post(f"{self.url}", data={})
+        self.assertEqual(result.status_code, 400)
+        self.assertEqual(models.Model.objects.all().count(), 1)
+```
+
+The setUp method here is not optimal because the line `self.model = factories.ModelFactory()` is going to be executed twice, meaning that it will create and destroy an object before running the first test, and then it's going to do it again for the other test (and again for all other tests in the TestCase).
```

Comment:
`create an object` ?
it doesn't destroy anything before the first test

Reply:
Not sure I understand. There's a rollback if another TestCase ran before, but if it's the first TestCase there's nothing being destroyed.

The point of that paragraph is to explain how setUp works, saying that it runs every time between tests, so there's this process of destruction and creation that usually is unnecessary and can be improved with setUpTestData.
Makes sense?